keymanapp/keyman
# chore(oem/fv): Add keyboard version and languages

This updates the keyboards.csv file used by the FirstVoices oem app to generate a keyboard list sorted by Regions.
The following columns are added: Keyboard Version, Language ID, and Language Name (each keyboard only has 1 associated language per kmp.json). These fields will be used by the upcoming feature of adding dictionary support for FV on Android 

The following Bash script generated the columns

```bash
##
## Parses kmp.json from fv_all.kmp and add the keyboard version and the 
## first associated language ID/name for each keyboard in the existing keyboards.csv
## Writes out to the target keyboards.csv
##
function updateLanguages {
  local KMP_FILE="kmp.json"
  if [ ! -f "$KMP_FILE" ]; then
    echo "ERROR: kmp.json not extracted"
    exit 1
  fi

  if [ -f $KEYBOARDS_CSV_TARGET ]; then
    echo "Cleaning existing keyboards.csv target"
    rm -f "${KEYBOARDS_CSV_TARGET}"
  fi

  echo "Parsing language info to add to keyboards.csv"

  # Existing column headers
  while IFS="," read -r SHORTNAME ID NAME REGION WEB_KEYBOARD_9_0
  do
    if [ "$SHORTNAME" != "Shortname" ]; then
      # Find first matching language info for each keyboard
      local KEYBOARD_INFO=$(cat "${KMP_FILE}" | "$JQ" -r ".keyboards[] | select(.id==\"$ID\")")
      local KEYBOARD_VERSION=$(echo ${KEYBOARD_INFO} | "$JQ" -r ".version")
      local LANGUAGE_INFO=$(echo ${KEYBOARD_INFO} | "$JQ" -r ".languages[0]")
      local LANGUAGE_ID=$(echo ${LANGUAGE_INFO} | "$JQ" -r ".id")
      local LANGUAGE_NAME=$(echo ${LANGUAGE_INFO} | "$JQ" -r ".name")

      # trim newline from WEB_KEYBOARD_9_0
      WEB_KEYBOARD_9_0=${WEB_KEYBOARD_9_0%$'\n'}
      WEB_KEYBOARD_9_0=${WEB_KEYBOARD_9_0%$'\r'}

      # Write out keyboard info
      echo "${SHORTNAME},${ID},${NAME},${REGION},${WEB_KEYBOARD_9_0},${KEYBOARD_VERSION},${LANGUAGE_ID},${LANGUAGE_NAME}" >> $KEYBOARDS_CSV_TARGET
    else
      # Copy header line to target
      echo "Shortname,ID,Name,Region,9.0 Web Keyboard,Version,Language ID,Language Name" >> $KEYBOARDS_CSV_TARGET
    fi
  done < $KEYBOARDS_CSV

  echo "Language info added to ${KEYBOARDS_CSV_TARGET}"
}
```

Tagging @madebybridget

## User Testing
* **TEST_ANDROID** 
1. Load the PR build of the OEM FirstVoices app for Android
2. On the Setup page, select a region --> any keyboard --> check the box
3. Back out until you're back at the Setup page
4. Enable FirstVoices as a System Keyboard
5. Launch a separate app (e.g. Chrome browser) and click in a text field
6. With the FirstVoices keyboard as a system keyboard, verify it appears and types

* **TEST_IOS**
1. Repeat the same test with the PR build of the OEM FirstVoices app for iOS
